parser/parse.go

```diff
@@ -260,6 +260,16 @@ func (b *Builder) AddDirTo(dir string, u *types.Universe) error {
 	return b.findTypesIn(canonicalizeImportPath(b.buildPackages[dir].ImportPath), u)
 }
 
+// DirectoryToPackage returns the package path given a dir. dir must be already
+// parsed.
+func (b *Builder) DirectoryToPackagePath(dir string) (string, error) {
```

Comment:
I still think my first version is better. I can give the method a better name, maybe AddDirectory (just to be different from AddDir). WDYT?